### PR TITLE
repo: error for packages with broken dependencies

### PR DESCRIPTION
**Pull request by kalikiana**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----
This change introduces a new error message instead of silently failing when a package has broken dependencies:

`The package ceph-common:armhf has unmet dependencies: python-requests`

Fixes: [bug 1686481](https://bugs.launchpad.net/snapcraft/+bug/1686481)

### CODE REVIEW

File: snapcraft/internal/repo/_deb.py [hunk 1]
```diff
@@ -221,9 +221,20 @@ def _mark_install(cls, apt_cache, package_names):
                 if version:
                     _set_pkg_version(apt_cache[name_arch], version)
                 apt_cache[name_arch].mark_install()
+                cls._verify_installable(apt_cache[name_arch])
             except KeyError:
                 raise errors.PackageNotFoundError(name)
 
+    @classmethod
+    def _verify_installable(cls, package: apt.Package):
```
[the remark this answers is not in the document]
Comment (kalikiana) [reply snapcraft/internal/repo/_deb.py]: In my understanding mark_installed resolves dependencies and sets marked_install to answer the question of "Can this be installed". But I have no strong opinion on the name really, so I renamed it.

**Comment by kalikiana** [comment]
Apparently I forgot to commit/ push the unit tests last night... sorry about that.

**Comment by kyrofa** [comment]
(although there's a test failure here FYI)

**Comment by kalikiana** [comment]
For the record, CI failed with `The contributor daniellimweesoong2000@gmail.com has not signed the CLA.`

**Pull request merged**